### PR TITLE
feat(lh-86970): add Terraform resource to add an existing tenant to MSP portal using API token

### PR DESCRIPTION
**Pull request by siddhuwarrier**
https://jira-eng-rtp3.cisco.com/jira/browse/LH-86970

### Description

1. Make the name field optional; require it only if api_token is not specified, and vice versa. This allows us to add existing tenants.
2. Add an existing tenant as follows:
```
resource "cdo_msp_managed_tenant" "existing_tenant" {
  api_token = "existing-tenant-api-token"
}
```

TODO: unit and acceptance tests

### CODE REVIEW

File: client/msp/tenants/add.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package tenants
+
+import (
+	"context"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/http"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/url"
+)
+
+func AddExistingTenantUsingApiToken(ctx context.Context, client http.Client, addInp MspAddExistingTenantInput) (*MspTenantOutput, *CreateError) {
+	client.Logger.Println("Creating tenant for CDO")
```
Comment (talhazi) [inline client/msp/tenants/add.go:10]: I suggest updating the comment, as we're only adding the tenant not creating it

File: client/msp/tenants/add.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+	var createOutp MspManagedTenantStatusInfo
+	if err := req.Send(&createOutp); err != nil {
+		return nil, &CreateError{Err: err}
+	}
+
+	return &createOutp.MspManagedTenant, nil
+}
+
+type StatusInfo struct {
```
Comment (talhazi) [inline client/msp/tenants/add.go:23]: Not sure, where are you using this?

**Review by talhazi — APPROVED** [review]
small ones, LGTM

File: client/msp/tenants/add.go (hunk 1 above)
Comment (siddhuwarrier) [reply client/msp/tenants/add.go:10]: Ouch, good catch, mate. Fixing.

**Pull request merged**